bird: drop bird1 package & modules; hardening

Mic92 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

reason: no longer maintained upstream.
No testing beyond running the nixos test has been conducted.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Mic92 committed:
bird1: drop package + modules

reason: no longer maintained upstream.

mweinelt commented on nixos/modules/services/networking/bird.nix:
```diff
@@ -3,103 +3,103 @@
+        ExecStop = "${pkgs.bird}/bin/birdc down";
+        CapabilityBoundingSet = [
+          "CAP_CHOWN"
+          "CAP_FOWNER"
+          "CAP_DAC_OVERRIDE"
```
Wow, those two are pretty rough. Bypassing filesystem permissions checks altogether seems like a bad move. I don't suppose you remember why these are required?

Mic92 replied:
Nope. I don't know a good reasons. Maybe we can drop it?

flokli replied:
It seems it's not required. I'd propose using the same set as https://github.com/archlinux/svntogit-packages/blob/packages/bird/trunk/bird.service#L20-L21 here. If the VM test still succeeds, we should be good.

mweinelt replied:
CAP_CHOWN and CAP_FOWNER are used to set permissions on the control socket. The test runs fine without CAP_DAC_OVERRIDE, so not sure why it would be required.

Mic92 replied:
done.

Mic92 replied:
It no longer would need root for permissions sockets but it still does for netlink sockets...

0x4A6F reviewed (changes requested):
Please also mention dropped package/service in release notes.

Mic92 commented:
Added changelog.

Mic92 committed:
nixos/bird: improve systemd hardening / capability set

aligned with https://github.com/archlinux/svntogit-packages/blob/packages/bird/trunk/bird.service#L20-L21

Mic92 committed:
bird: change runtstatedir to systemd convention

0x4A6F commented on nixos/doc/manual/release-notes/rl-2205.section.md:
```diff
@@ -176,6 +176,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - `tilp2` was removed together with its module
 
+- `bird1` and its module has `services.bird` as well as `services.bird6` has been removed. Upgrade to `services.bird2`.
```
Maybe a native speaker has a better suggestion.
```suggestion
- `bird1` and its modules `services.bird` as well as `services.bird6` have been removed. Upgrade to `services.bird2`.
```

0x4A6F commented:
Thanks. Apart from that minor nitpick `services.bird2` still works with my bgp-peers with your latest commit.

flokli commented on nixos/doc/manual/release-notes/rl-2205.section.md:
```diff
@@ -176,6 +176,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 - `tilp2` was removed together with its module
 
+- `bird1` and its module has `services.bird` as well as `services.bird6` has been removed. Upgrade to `services.bird2`.
+
```
We should probably also add the fact bird is running as root again, so it can drop privileges, instead of running with the scary capabilities.

Mic92 replied:
This was already the case before in fact.

Mic92 replied:
see https://github.com/NixOS/nixpkgs/pull/161193/files#diff-7b8f5c81bf510eb2bb574f1374914deebdafec8f5c001f9ea9983312438fa1e7L78

0x4A6F reviewed (approved):
Thanks.

chkno commented:
`git bisect` says a5276e1fbdb1773554fb882ad4432116b9b4d918 broke `nix-build -A bird nixos/tests/prometheus-exporters.nix`

[Failed build logs](https://hydra.nixos.org/build/171140708) show the error `dial unix /var/run/bird.ctl: connect: no such file or directory`

flokli commented:
fixed in https://github.com/NixOS/nixpkgs/pull/166875.
